lib/Core/Executor.cpp

```diff
@@ -905,14 +905,21 @@ Executor::fork(ExecutionState &current, ref<Expr> condition, bool isInternal) {
       }
     }
 
+    // We don't add a constraint explicitly yet
```

Comment:
@ccadar Yes, I agree with you especially with the fact that it is a general problem with those heuristics. And as long as we don't have a progress indicator (like N instructions or added constraints) to reschedule, there are scenarios it will get stuck.
Maybe we should add this as a point to our wishlist.

@wuestholz I'll open the original patch as a separate PR. And we can eventually close this one. Meanwhile, you can still use this one if it helps solving your specific problem. The good thing of the  previous one is that it is less invasive.